pipelines/_common.toml

```diff
@@ -0,0 +1,56 @@
+# Commmon layout components for POPROX pipelines
+
+[meta]
+name = "NRMS"
+
+# Define convenient aliases for components so we do not need to re-type their
+# full names everywhere.  Inherting configurations can override these to change
+# the component without overwriting it!
+[definitions]
+# the name of the final personalized ranking component
+ranker = "ranker"
+# number of recommendation slots to configure
+N = 10
+
+[inputs.candidate]
+type = "poprox_concepts.domain.article.ArticleSet"
+[inputs.profile]
+type = "poprox_concepts.domain.article.ArticleSet"
+[inputs.clicked]
+type = "poprox_concepts.domain.article.InterestProfile"
+
+[components.candidate-embedder]
+code = "poprox_recommender.components.embedders.article:NRMSArticleEmbedder"
+config = { model_path = "nrms-mind/news_encoder.safetensors" }
+inputs = { article_set = "candidate" }
+
+[components.history-embedder]
+code = "poprox_recommender.components.embedders.article:NRMSArticleEmbedder"
+config = { model_path = "nrms-mind/news_encoder.safetensors" }
+inputs = { article_set = "candidate" }
```

Comment:
An idea: to avoid loading the embedder twice, we could add "reuse" capabilities:

```toml
[components.history-embedder]
reuse = "candidate-embedder"
inputs = { article_set = "candidate" }
```

Reply:
This bit will require new pipeline logic, not just a configuration file transform layer, but is entirely feasible.